libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/EdgeToEdgeUtil.kt

```diff
@@ -0,0 +1,36 @@
+package com.salesforce.androidsdk.ui
+
+import android.graphics.drawable.LayerDrawable
+import android.os.Build.VERSION_CODES.UPSIDE_DOWN_CAKE
+import android.view.View
+import androidx.activity.enableEdgeToEdge
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.res.ResourcesCompat
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.core.view.updatePadding
+import com.salesforce.androidsdk.R.drawable.sf__fix_status_bar
+import com.salesforce.androidsdk.R.id.sf__status_bar_background
+
+// TODO: Remove this in 13.0 after rewriting screens in compose.
+internal fun AppCompatActivity.fixEdgeToEdge(view: View) {
+    if (application.applicationInfo.targetSdkVersion > UPSIDE_DOWN_CAKE) {
+        enableEdgeToEdge()
+        ViewCompat.setOnApplyWindowInsetsListener(view) { listenerView, windowInsets ->
+            val insets = windowInsets.getInsets(
+                WindowInsetsCompat.Type.systemBars() or WindowInsetsCompat.Type.displayCutout()
+            )
+
+            // Fix screens being drawn behind status and navigation bars
+            listenerView.updatePadding(insets.left, insets.top, insets.right, insets.bottom)
+
+            // Fix transparent status bar not matching action bar
+            val background = ResourcesCompat.getDrawable(resources, sf__fix_status_bar, null)
```

Comment:
Is there a chance background could be null? Should we check for null? E.g. if they subclass our activity and use a different layout.

Reply:
"background" is the new [sf__fix_status_bar.xml](https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2624/files/5a859840bbab9d2ccd2f7a5f632982a9d9cc2162#diff-d3e42c86dfab7e3f197037eb5cd45c220cce7298278a935c00277d3a038010e5) drawable added in this PR.  I set it as the background of the activity later with `view.setBackgroundResource(sf__fix_status_bar)`.  

The drawable is split in half, the top being `sf__primary_color` and the bottom is transparent.  I am getting a reference to it here so I can set the height of the top to the height of the status bar.  This mimics the old look.